legion/legion/template.py

```diff
@@ -0,0 +1,230 @@
+#
+#    Copyright 2018 EPAM Systems
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+"""
+Template system that uses courutines
+"""
+import logging
+import os
+import os.path
+import importlib
+
+import asyncio
+from asyncio.tasks import Task
+
+from jinja2 import Environment, FileSystemLoader, Undefined
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LegionTemplate:
```

Comment:
Does not look like a template to me. More like an engine or something.

Reply:
OK, will be renamed to LegionTemplateEngine